src/utils/run-build.mjs

```diff
@@ -123,6 +123,7 @@ export const runNetlifyBuild = async ({ command, env = {}, options, settings, ti
     await devCommand({
       command: undefined,
       useStaticServer: true,
+      dist: options.dir ? undefined : netlifyConfig?.build?.publish,
```

Comment:
Agree it feels odd. But, always sending `dist` will effectively override all settings that come out of `detectServerSettings`, so i'd prefer not to do that.
Now that I think about it, `undefined` will also override `dist` because it's being destructured in `devCommand` - gonna need to change this.